payments-core/src/main/java/com/stripe/android/payments/bankaccount/ui/CollectBankAccountViewModel.kt

```diff
@@ -58,20 +54,40 @@ internal class CollectBankAccountViewModel @Inject constructor(
     private suspend fun createFinancialConnectionsSession() {
         when (val configuration = args.configuration) {
             is CollectBankAccountConfiguration.USBankAccount -> when (args) {
-                is ForPaymentIntent -> createFinancialConnectionsSession.forPaymentIntent(
-                    publishableKey = args.publishableKey,
-                    stripeAccountId = args.stripeAccountId,
-                    clientSecret = args.clientSecret,
-                    customerName = configuration.name,
-                    customerEmail = configuration.email
-                )
-                is ForSetupIntent -> createFinancialConnectionsSession.forSetupIntent(
-                    publishableKey = args.publishableKey,
-                    stripeAccountId = args.stripeAccountId,
-                    clientSecret = args.clientSecret,
-                    customerName = configuration.name,
-                    customerEmail = configuration.email
-                )
+                is CollectBankAccountContract.Args.ForDeferredPaymentIntent ->
+                    createFinancialConnectionsSession.forDeferredPayments(
+                        publishableKey = args.publishableKey,
+                        stripeAccountId = args.stripeAccountId,
+                        elementsSessionId = args.elementsSessionId,
+                        customerId = args.customerId,
+                        amount = args.amount,
+                        currency = args.currency
+                    )
+                is CollectBankAccountContract.Args.ForDeferredSetupIntent ->
+                    createFinancialConnectionsSession.forDeferredPayments(
+                        publishableKey = args.publishableKey,
+                        stripeAccountId = args.stripeAccountId,
+                        elementsSessionId = args.elementsSessionId,
+                        customerId = args.customerId,
+                        amount = null,
+                        currency = null
+                    )
```

Comment:
we'll be using the same endpoint for both cases right?

Reply:
Yes that is correct!